tests/test_content_loader.py

```diff
@@ -2033,13 +2029,83 @@ def test_caching_of_messages(self, mock_read_yaml):
         mock_read_yaml.assert_called_once_with('content/frameworks/g-cloud-7/messages/index.yml')
 
     def test_load_message_raises(self, mock_read_yaml):
-
         mock_read_yaml.side_effect = IOError
         messages = ContentLoader('content/')
 
         with pytest.raises(ContentNotFoundError):
             messages.load_messages('not-a-framework', ['index'])
 
+    def test_get_metadata(self, mock_read_yaml):
+        mock_read_yaml.return_value = {
+            'field_one': 'value_one',
+            'field_two': 'value_two',
+        }
+        metadata = ContentLoader('content/')
+        metadata.load_metadata('g-cloud-7', ['index'])
+
+        assert metadata.get_metadata('g-cloud-7', 'index') == ContentMetadata({
+            'field_one': 'value_one',
+            'field_two': 'value_two',
+        })
+
+        assert metadata.get_metadata('g-cloud-7', 'index').field_one == 'value_one'
+
+    def test_get_metadata_with_key(self, mock_read_yaml):
+        mock_read_yaml.return_value = {
+            'field_one': 'value_one',
+            'field_two': 'value_two',
+        }
+        metadata = ContentLoader('content/')
+        metadata.load_metadata('g-cloud-7', ['index'])
+
+        assert metadata.get_metadata('g-cloud-7', 'index', 'field_one') == 'value_one'
+
+    def test_load_metadata_wraps_nested_fields_with_template_field(self, mock_read_yaml):
+        mock_read_yaml.return_value = {
+            'source_framework': 'g-cloud-6',
+            'questions_to_copy': ['question-1', 'question-2']
+        }
+        metadata = ContentLoader('content/')
+        metadata.load_metadata('g-cloud-7', ['copy_services'])
+
+        assert metadata.get_metadata('g-cloud-7', 'copy_services') == ContentMetadata({
+            'source_framework': 'g-cloud-6',
+            'questions_to_copy': ['question-1', 'question-2']
+        })
+
+        assert metadata.get_metadata('g-cloud-7', 'copy_services').source_framework == 'g-cloud-6'
+
+    def test_load_metadata_argument_types(self, mock_read_yaml):
+        mock_read_yaml.return_value = {}
+        metadata = ContentLoader('content/')
+
+        with pytest.raises(TypeError):
+            metadata.load_metadata('g-cloud-7', 'index')  # blocks argument must be a list
```

Comment:
Could check the error message instead of having the comment.

Reply:
Done.